src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -118,10 +119,10 @@ const DateTimeBlock : React.FC<IProps> = ({
   setDateCallback = ()=>{},
 }) => {
 
-  const convertHours = (date.getHours()).toString();
-  const convertMinutes = date.getMinutes().toString();
-  const [displayHours, setDisplayHours] = useState<string>(convertHours === '0'?'00':convertHours);
-  const [displayMinutes, setDisplayMinutes] = useState<string>(convertMinutes === '0'?'00':convertMinutes);
+  const convertHours = (date?.getHours()).toString();
+  const convertMinutes = (date?.getMinutes()).toString();
```

Comment:
What happens if `date` is `undefined`? Currently it seems this prop is optional and then as this is used to set the default state of the `displayHours` and `displayMinutes`. However `getFormattedTime` currently needs `value` to be a string.

It seems like there is an error case here that needs fixing. I would have thought Typescript would be giving you an error for this but if I've misunderstood something, please let me know!

Reply:
I have add one extra condition in getFormattedTime() . When value is undefined it will return '00'

Reply:
The `getFormattedTime` is still set to require a string, so in the instance of undefined it's going to error rather than reach the new `else` state.

When you compile are there any errors from Typescript? I'm surprised it's not catching that one.

Reply:
Yes there is no error. I have checked.